libnymea-core/tagging/tagsstorage.h

```diff
@@ -0,0 +1,79 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ *                                                                         *
+ *  Copyright (C) 2018 Michael Zanetti <michael.zanetti@guh.io>            *
+ *                                                                         *
+ *  This file is part of nymea.                                            *
+ *                                                                         *
+ *  nymea is free software: you can redistribute it and/or modify          *
+ *  it under the terms of the GNU General Public License as published by   *
+ *  the Free Software Foundation, version 2 of the License.                *
+ *                                                                         *
+ *  nymea is distributed in the hope that it will be useful,               *
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of         *
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          *
+ *  GNU General Public License for more details.                           *
+ *                                                                         *
+ *  You should have received a copy of the GNU General Public License      *
+ *  along with nymea. If not, see <http://www.gnu.org/licenses/>.          *
+ *                                                                         *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#ifndef TAGSSTORAGE_H
+#define TAGSSTORAGE_H
+
+#include "tag.h"
+#include "typeutils.h"
+
+#include <QObject>
+#include <QVector>
+
+class DeviceManager;
+
+namespace nymeaserver {
+
+class RuleEngine;
+
+class TagsStorage : public QObject
+{
+    Q_OBJECT
+public:
+    enum TagError {
+        TagErrorNoError,
+        TagErrorDeviceNotFound,
+        TagErrorRuleNotFound,
+        TagErrorTagNotFound,
+        TagErrorTagExists
```

Comment:
Is there a case where the tag content is messed up? Would an error like `TagErrorInvalidParameters` or `TagErrorInconsistent` make sense?